Angulartics-piwik plugin added

florentbruel opened the pull request:
Angulartics does not provide by default the piwik module (see
https://github.com/angulartics/angulartics). angulartics-piwik plugin
bower component added.
Same requirements for other plugins defined in w20 add-on, if this pull request is valide, I can update the issue with all the plugins currently supported by w20-extras / analytics

florentbruel committed:
Angulartics-piwik plugin added

Angulartics does not provide by default the piwik module (see
https://github.com/angulartics/angulartics). angulartics-piwik plugin
bower component added.

florentbruel committed:
Update analytics.js

florentbruel committed:
Update analytics.js

adrienlauer commented:
Since providers are optional, I think we should not force the dependency upon projets using w20-extra. People needing the dependency will have to add it explicitly. Defining the necessary path mapping and shim for those plugins is a good idea though.

We may have a problem with the w20-bridge add-on, where those dependencies will be missing. Maybe we could add them in the bridge bower.json file (https://github.com/seedstack/w20-bridge-addon/blob/master/web-extras/src/main/js/bower.json) so they are readily available for projects using the bridge.

@florentbruel: if you agree, you can do the following things so I can merge this PR:

* Remove the piwik bower dependency,
* Add all necessary code and config (mainly path mapping and shim) for all Angulartics plugins so they can work if they are added explicitly by projects.

Later on, we can add the bower dependencies for plugins in the w20-bridge.

florentbruel committed:
fallback for piwik initialization

fallback for piwik initialization may jsUrl not be reachable.
Fallback : piwik conf not set, and requirejs dependency on $window._paq not executed for the module to be initialized without error.
A console warn is written, so that the developers can be fully aware of the application state.

florentbruel committed:
Update bower.json

florentbruel committed:
Adding all angulartics supported providers

require js shim and path configurations

florentbruel committed:
adding new providers supported by angulartics

florentbruel committed:
bad spelling

florentbruel commented:
@adrienlauer bower dependency to angulartics-piwik has been removed, and all path / shim configurations have been added to ease the provider integration.
